src/engine.cc

```diff
@@ -138,9 +138,9 @@ void EngineController::PopulateOptions(OptionsParser* options) {
 }
 
 SearchLimits EngineController::PopulateSearchLimits(int ply, bool is_black,
-                                                    const GoParams& params) {
+    const GoParams& params, std::chrono::steady_clock::time_point start_time) {
   SearchLimits limits;
-  limits.time_ms = params.movetime;
+  limits.movetime = params.movetime;
```

Comment:
I think rather than this - limits.search_deadline should be set to start_time + params.movetime in the return on line 154, if infinite is not true.

Reply:
I had it like that initially, but according to the uci spec, the meaning of `movetime` is "search exactly x mseconds" so I changed it.

Reply:
At least [cutechess](https://github.com/cutechess/cutechess/blob/25e6ed6204698932b6db577316d2c088d2a0a0a6/projects/lib/src/uciengine.cpp#L266) uses `movetime` so this requires further analysis.

Reply:
Given the existing lack of lag time subtraction from movetime - I think we can say this is at least not a regression and so leave it to a follow up PR at some future point.